aws-common/src/main/java/org/apache/druid/common/aws/FileSessionCredentialsProvider.java

```diff
@@ -24,27 +24,25 @@
 import com.amazonaws.auth.AWSSessionCredentials;
 import org.apache.druid.java.util.common.concurrent.Execs;
 
-import java.io.File;
-import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.util.Properties;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
 public class FileSessionCredentialsProvider implements AWSCredentialsProvider
 {
-  private final String sessionCredentials;
-  private volatile String sessionToken;
-  private volatile String accessKey;
-  private volatile String secretKey;
+  private final String sessionCredentialsFile;
+  private AWSSessionCredentials awsSessionCredentials;
```

Comment:
Sorry for leaving a comment after LGTM, but is this needed to be volatile?

Reply:
It doesn't need to be volatile. From JMM point of view, volatile on this field changes anything and doesn't provide any extra guarantees.

Reply:
I'll leave this as a comment in code though.

Reply:
Right, thanks.